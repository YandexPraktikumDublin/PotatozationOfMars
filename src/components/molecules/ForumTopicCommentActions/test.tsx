import React from 'react'
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'
import { Store } from 'redux'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'

import { ForumTopicCommentActions } from '.'

const mockStore = configureStore([])

describe('<ForumTopicCommentActions />', () => {
  const comment = {
    id: 1,
    content: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit.',
    userId: 1,
    user: {
      id: 1,
      login: 'IvanIvanov',
      name: 'Ivan Ivanov',
      createdAt: '01.12.20',
      updatedAt: '01.12.20'
    },
    topicId: 1,
    reactions: [
      {
        id: 1,
        content: 'ðŸ˜‚',
        userId: 1,
        user: {
          id: 1,
          login: 'IvanIvanov',
          name: 'Ivan Ivanov',
          createdAt: '01.12.20',
          updatedAt: '01.12.20'
        },
        commentId: 2,
        createdAt: '01.12.20',
        updatedAt: '01.12.20'
      }
    ],
    createdAt: '01.12.20',
    updatedAt: '01.12.20'
  }

  let store: Store

  const initialState = {
    comment
  }

  beforeEach(() => {
    store = mockStore(initialState)
  })

  it('should renders correct <ForumTopicCommentActions />', () => {
    const wrapper = shallow(
      <Provider store={store}>
        <ForumTopicCommentActions comment={comment} />
      </Provider>
    )

    expect(toJSON(wrapper)).toMatchSnapshot()
  })
})
