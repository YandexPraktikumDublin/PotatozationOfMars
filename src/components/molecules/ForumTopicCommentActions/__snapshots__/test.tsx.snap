// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ForumTopicCommentActions /> should renders correct <ForumTopicCommentActions /> 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <ForumTopicCommentActions
    comment={
      Object {
        "content": "Lorem ipsum dolor sit amet, consectetur adipiscing elit.",
        "createdAt": "01.12.20",
        "enjoyer": Object {
          "createdAt": "01.12.20",
          "id": 1,
          "login": "IvanIvanov",
          "name": "Ivan Ivanov",
          "updatedAt": "01.12.20",
        },
        "enjoyerId": 1,
        "id": 1,
        "reactions": Array [
          Object {
            "commentId": 2,
            "content": "ðŸ˜‚",
            "createdAt": "01.12.20",
            "enjoyer": Object {
              "createdAt": "01.12.20",
              "id": 1,
              "login": "IvanIvanov",
              "name": "Ivan Ivanov",
              "updatedAt": "01.12.20",
            },
            "enjoyerId": 1,
            "id": 1,
            "updatedAt": "01.12.20",
          },
        ],
        "topicId": 1,
        "updatedAt": "01.12.20",
      }
    }
  />
</ContextProvider>
`;
